src/main/java/com/redhat/devtools/intellij/quarkus/module/QuarkusCodeEndpointChooserStep.java

```diff
@@ -15,42 +15,57 @@
 import com.intellij.ide.util.PropertiesComponent;
 import com.intellij.ide.util.projectWizard.ModuleWizardStep;
 import com.intellij.ide.util.projectWizard.WizardContext;
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.application.ModalityState;
 import com.intellij.openapi.options.ConfigurationException;
 import com.intellij.openapi.progress.EmptyProgressIndicator;
+import com.intellij.openapi.progress.ProgressManager;
 import com.intellij.openapi.ui.ComponentWithBrowseButton;
 import com.intellij.openapi.ui.Messages;
 import com.intellij.openapi.util.Comparing;
+import com.intellij.openapi.util.Disposer;
 import com.intellij.ui.HyperlinkLabel;
-import com.intellij.ui.IdeBorderFactory;
 import com.intellij.ui.TextFieldWithStoredHistory;
 import com.intellij.ui.components.JBLabel;
 import com.intellij.ui.components.JBRadioButton;
+import com.intellij.util.ui.AsyncProcessIcon;
 import com.intellij.util.ui.FormBuilder;
 import com.intellij.util.ui.JBUI;
 import com.intellij.util.ui.components.BorderLayoutPanel;
 import com.redhat.devtools.intellij.quarkus.QuarkusConstants;
 
-import javax.swing.ButtonGroup;
-import javax.swing.JComponent;
-import javax.swing.JPanel;
-import java.awt.BorderLayout;
+import javax.swing.*;
+import java.awt.*;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
-import java.io.IOException;
 import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 
 import static com.redhat.devtools.intellij.quarkus.QuarkusConstants.QUARKUS_CODE_URL;
+import static com.redhat.devtools.intellij.quarkus.module.QuarkusModelRegistry.DEFAULT_TIMEOUT_IN_SEC;
 
-public class QuarkusCodeEndpointChooserStep extends ModuleWizardStep {
+public class QuarkusCodeEndpointChooserStep extends ModuleWizardStep implements Disposable {
     private static final String LAST_ENDPOINT_URL = "quarkus.code.endpoint.url.last";
     private static final String ENDPOINT_URL_HISTORY = "quarkus.code.endpoint.url.history";
     private final WizardContext wizardContext;
     private final JBRadioButton defaultRadioButton = new JBRadioButton("Default:", true);
     private final JBRadioButton customRadioButton = new JBRadioButton("Custom:", false);
     private final TextFieldWithStoredHistory endpointURL = new TextFieldWithStoredHistory(ENDPOINT_URL_HISTORY);
     private final ComponentWithBrowseButton<TextFieldWithStoredHistory> customUrlWithBrowseButton;
+    private final JPanel component;
+
+    private Future<QuarkusModel> loadingRequest = null;
+
+    private QuarkusModel streams = null;
+
+    private AsyncProcessIcon spinner = new AsyncProcessIcon("Loading Quarkus data...");
```

Comment:
In my PR for Quarkus Explorer I provide a QuarkusBundle for messages https://github.com/redhat-developer/intellij-quarkus/pull/1178/files#diff-a1cdf4cad7aad393c6e211ef96ba1bdbff1119ca407aae4de88086ebcb024277 I think it should be nice to use it for wizard label components

Reply:
done